code/modules/recycling/conveyor2.dm

```diff
@@ -265,3 +301,85 @@
 		if(S.id == src.id)
 			S.position = position
 			S.update()
+
+/obj/machinery/conveyor_switch/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/crowbar))
+		var/obj/item/conveyor_switch_construct/C = new/obj/item/conveyor_switch_construct(src.loc)
+		C.id = id
+		transfer_fingerprints_to(C)
+		user << "<span class='notice'>You deattach the conveyor switch.</span>"
+		qdel(src)
+
+	else if(istype(I, /obj/item/device/multitool))
+		update_multitool_menu(user)
+		return 1
+
+
+/obj/machinery/conveyor_switch/multitool_menu(var/mob/user, var/obj/item/device/multitool/P)
+	return {"
+ 	<ul>
+ 	<li><b>One direction only:</b> <a href='?src=\ref[src];toggle_logic=1'>[convdir ? "On" : "Off"]</a></li>

```

Comment:
Now it works, but the UI doesn't update when you click the button - so it appears to do nothing. Is there something I can add in there to get the thing to refresh when you click it?